SimCore/include/SimCore/G4User/PtrRetrieval.h

```diff
@@ -0,0 +1,104 @@
+#ifndef SIMCORE_PTRRETRIEVAL_H
+#define SIMCORE_PTRRETRIEVAL_H
+
+//----------------//
+//   C++ StdLib   //
+//----------------//
+#include <iostream>
+#include <string>
+
+//------------//
+//   Geant4   //
+//------------//
+#include "G4Gamma.hh"
+#include "G4ParticleDefinition.hh"
+#include "G4PhysicalVolumeStore.hh"
+#include "G4ProcessManager.hh"
+#include "G4Region.hh"
+#include "G4RegionStore.hh"
+#include "G4VPhysicalVolume.hh"
+#include "G4LogicalVolume.hh"
+#include "G4LogicalVolumeStore.hh"
+
+namespace simcore {
+namespace g4user {
+namespace ptrretrieval {
+
+/**
+ * @brief Retrieve a specific process for a given particle type.
+ * @param particle Pointer to the G4ParticleDefinition object.
+ * @param processName Name of the process to find.
+ * @return Pointer to the process if found, nullptr otherwise.
+ */
+inline const G4VProcess* getProcess(const G4ParticleDefinition* particle,
+                                    const std::string& processName) {
+  if (!particle) return nullptr;
+
+  const auto manager = particle->GetProcessManager();
+  if (!manager) return nullptr;
+
+  const auto processes = manager->GetProcessList();
+  for (int i = 0; i < processes->size(); ++i) {
+    const auto process = (*processes)[i];
+    if (process->GetProcessName().find(processName) != std::string::npos) {
+      return process;
+    }
+  }
+  std::cerr << "[WARNING] Process '" << processName << "' not found.\n";
```

Comment:
Dont use cerr use the ldmx logging
```suggestion
  ldmx_log(warn) << "Process '" << processName << "' not found.";
```

Reply:
Talking with Liam I understand this might not be simple, as this is a namespace. Should we not throw the warning in the user class end? i.e. if the `getProcess` returned a `nullptr` then have the warning in the respective filter? @EinarElen @tomeichlersmith

Reply:
I vote to have the warning in the user class end although its not technically difficult to support logging within a namespace. The only requirement for `ldmx_log` to function is for a `framework::logging::logger` named `theLog_` to be present within scope. A namespace-level variable _could_ then support logging for all of the functions under one "channel", but I think it makes more sense to have the warning be in the user classes.